apps/src/code-studio/teacherPanelHelpers.js

```diff
@@ -24,67 +16,17 @@ export function renderTeacherPanel(
   const div = document.createElement('div');
   div.setAttribute('id', 'teacher-panel-container');
 
-  queryStudentsForSection(store);
-  queryLockStatus(store, scriptId, pageType);
-
   ReactDOM.render(
     <Provider store={store}>
-      <TeacherPanel unitName={scriptName} pageType={pageType} />
+      <TeachersOnly>
+        <TeacherPanel
+          unitName={scriptName}
+          pageType={pageType}
+          scriptId={scriptId}
+        />
+      </TeachersOnly>
     </Provider>,
     div
   );
   document.body.appendChild(div);
 }
-
-/**
- * Query the server for lock status of this teacher's students
- * @returns {Promise} when finished
- */
-function queryLockStatus(store, scriptId, pageType) {
```

Comment:
Could you elaborate a bit on the motivation for moving this logic to TeacherPanel?  Was there a technical reason the code needed to be moved?

Reply:
Yes, I only want these requests to be made if the user is a teacher and we're definitely rendering the teacher panel. teacherPanelHelpers.renderTeacherPanel will be called even if the user is not a teacher. This is somewhat hard to explain in a paragraph. Let me know if you want to hop on a call and talk through the pros/cons of other approaches.

Reply:
I think I get it now, thanks for the explanation!